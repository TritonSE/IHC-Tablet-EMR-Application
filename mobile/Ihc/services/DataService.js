/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeDataServices === 'true') ? fakeLocalData : localDataService;
let serverData = (config.fakeDataServices === 'true') ? fakeServerData : serverDataService;
/*
let localData = (config.fakeDataServices === 'true' || global.__TEST__) ? fakeLocalData : localDataService;
let serverData = (config.fakeDataServices === 'true' || global.__TEST__) ? fakeServerData : serverDataService;
*/
export {localData, serverData};
