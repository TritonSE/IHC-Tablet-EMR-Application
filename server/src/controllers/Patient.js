import PatientModel from '../models/Patient';
import SoapModel from '../models/Soap';
import StatusModel from '../models/Status';
import TriageModel from '../models/Triage';
import DrugUpdateModel from '../models/DrugUpdate';

const PatientController = {
  GetPatient: function(req, res){
    PatientModel.findOne({key: req.params.key}, function(err, patient) {
      if(!patient) {
        err = new Error("Patient with key " + req.params.key + " doesn't exist");
      }
      if(err) {
        res.json({status: false, error: err.message});
        return;
      }
      res.json({status: true, patient: patient});
    });
  },
  GetPatients: function(req, res){
  },
  CreatePatient: function(req, res){
    // Check that no patient with that key exists
    PatientModel.findOne({key: req.body.patient.key}, function(err, patient) {
      if (patient) {
        // Hopefully shouldn't happen, but would be rare enough to not worry
        // about
        err = new Error("Patient already exists with that name and birthday. Use a different name");
      }
      if(err) {
        res.json({status: false, error: err.message});
        return;
      }
    });

    const patient = PatientModel.create(req.body.patient, function (err) {
      if(err) {
        res.json({status: false, error: err.message});
        return;
      }
      res.json({status: true});
    })
  },
  UpdatePatient: function(req, res){
  },
  GetUpdates: function(req, res){
  },
  GetSoap: function(req, res){
  },
  GetStatus: function(req, res){
  },
  GetTriage: function(req, res){
  },
  GetDrugUpdates: function(req, res){
  },
  UpdateSoap: function(req, res){
  },
  UpdateStatus: function(req, res){
  },
  UpdateTriage: function(req, res){
  },
  UpdateDrugUpdates: function(req, res){
  }
};

module.exports = PatientController;
