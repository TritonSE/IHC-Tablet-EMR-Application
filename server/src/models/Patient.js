import mongoose from 'mongoose';

const Schema = mongoose.Schema;
const PatientSchema = Schema({
  key: String, // A unique identifier created from front end
  firstName: String,
  fatherName: String, // last name
  motherName: String, // last name
  birthday: String,
  gender: Number, // 1 = boy, 2 = girl, 0 = undefined
  phone: String,
  motherHeight: Number,
  fatherHeight: Number,
  medications: [{ type: Schema.Types.ObjectId, ref: 'DrugUpdate' }],
  soaps: [{ type: Schema.Types.ObjectId, ref: 'Soap' }],
  triages: [{ type: Schema.Types.ObjectId, ref: 'Triage' }],
  statuses: [{ type: Schema.Types.ObjectId, ref: 'Status' }],
  lastUpdated: Number // timestamp
});
const PatientModel = mongoose.model('Patient', PatientSchema);
module.exports = PatientModel;
