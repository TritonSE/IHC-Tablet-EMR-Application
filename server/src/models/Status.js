import mongoose from 'mongoose';

const Schema = mongoose.Schema;
const StatusSchema = Schema({
  patientKey: String,
  name: String,
  birthday: String,
  date: String,
  active: Boolean,
  checkinTime: Number, // timestamp, milliseconds
  triageCompleted: Number, // timestamps for when completed
  doctorCompleted: Number,
  pharmacyCompleted: Number,
  notes: String,
  lastUpdated: Number // timestamp
});
const StatusModel = mongoose.model('Status', StatusSchema);
module.exports = StatusModel;
