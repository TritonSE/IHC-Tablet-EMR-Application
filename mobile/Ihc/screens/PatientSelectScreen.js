import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView,
} from 'react-native';
import {localData, serverData} from '../services/DataService';
import {stringDate} from '../util/Date';
import PatientTable from '../components/PatientTable';
import Container from '../components/Container';
import {downstreamSyncWithServer} from '../util/Sync';

class PatientSelectScreen extends Component<{}> {
  /*
   * Redux props:
   * loading: boolean
   */
  constructor(props) {
    super(props);

    this.state = {
      rows: [],
    };
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  convertStatusesToRows(statuses) {
    const columnOrder = ['name', 'birthday', 'checkinTime', 'triageCompleted',
      'doctorCompleted', 'pharmacyCompleted', 'notes', 'patientKey'];

    // Sort statuses by checkin time for now
    statuses.sort( (status1, status2) => status1.checkinTime - status2.checkinTime );

    const toReturn = statuses.map((obj) => columnOrder.map( (key) => obj[key] ));
    return toReturn;
  }

  // Sync up tablet first with server before grabbing statuses
  syncAndLoadPatients = () => {
    this.props.setCurrentPatientKey(null);
    this.props.setLoading(true);
    this.props.isUploading(false);
    this.props.clearMessages();

    // Load local data in beginning to display even if sync doesn't work
    const today = stringDate(new Date());
    const oldStatuses = localData.getStatuses(today);
    const oldRowData = this.convertStatusesToRows(oldStatuses);
    this.setState({rows: oldRowData});

    downstreamSyncWithServer()
      .then((failedPatientKeys) => {
        // View README: Handle syncing the tablet, point 3 for explanation
        if(this.props.loading) {
          if(failedPatientKeys.length > 0) {
            throw new Error(`${failedPatientKeys.length} patients didn't properly sync.`);
          }
          const newStatuses = localData.getStatuses(today);
          const newRowData = this.convertStatusesToRows(newStatuses);

          this.setState({rows: newRowData});
          this.props.setLoading(false);
        }
      })
      .catch(err => {
        if(this.props.loading) {
          this.props.setLoading(false);
          this.props.setErrorMessage(err.message);
        }
      });
  }

  // Reload table after moving back to table
  // Replaces componentDidMount() because this will be called around the same
  // time
  onNavigatorEvent(event) {
    if (event.id === 'willAppear') {
      this.syncAndLoadPatients();
    }
  }

  // patient is an array containing one row of data from the PatientTable
  // PatientKey stored in index 7
  goToPatient = (patient) => {
    this.props.setCurrentPatientKey(patient[7]);
    this.props.navigator.push({
      screen: 'Ihc.PatientHomeScreen',
      title: patient.name,
      passProps: { name: patient[0] }
    });
  }

  // Update the statusObj with notes from the modal
  saveModal = (patientKey, notes) => {
    let statusObj = {};
    try {
      statusObj = localData.updateStatus(patientKey, stringDate(new Date()),
        'notes', notes);
    } catch(e) {
      this.props.setErrorMessage(e.message);
      return;
    }

    this.props.setLoading(true);
    this.props.isUploading(true);
    this.props.setCurrentPatientKey(patientKey);

    serverData.updateStatus(statusObj)
      .then( () => {
        // View README: Handle syncing the tablet, point 3 for explanation
        if(this.props.loading) {
          // if successful, then reload screen (which closes modal too)
          this.syncAndLoadPatients();

          this.props.setLoading(false);
          this.props.setSuccessMessage('Saved successfully');
        }
      })
      .catch( (e) => {
        if(this.props.loading) {
          localData.markPatientNeedToUpload(patientKey);

          this.props.setLoading(false, true);
          this.props.setErrorMessage(e.message);
        }
      });
  }

  render() {
    return (
      <Container>
        <Text style={styles.title}>
          Select a Patient
        </Text>

        <ScrollView contentContainerStyle={styles.tableContainer} horizontal>
          <PatientTable
            rows={this.state.rows}
            goToPatient={this.goToPatient}
            saveModal={this.saveModal}
          />
        </ScrollView>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  tableContainer: {
    width: '100%'
  },
});

// Redux
import { setLoading, setErrorMessage, setSuccessMessage, clearMessages, isUploading, setCurrentPatientKey } from '../reduxActions/containerActions';
import { connect } from 'react-redux';

const mapStateToProps = state => ({
  loading: state.loading
});

const mapDispatchToProps = dispatch => ({
  setLoading: (val,showRetryButton) => dispatch(setLoading(val, showRetryButton)),
  setErrorMessage: val => dispatch(setErrorMessage(val)),
  setSuccessMessage: val => dispatch(setSuccessMessage(val)),
  clearMessages: () => dispatch(clearMessages()),
  isUploading: val => dispatch(isUploading(val)),
  setCurrentPatientKey: key => dispatch(setCurrentPatientKey(key))
});

export default connect(mapStateToProps, mapDispatchToProps)(PatientSelectScreen);
