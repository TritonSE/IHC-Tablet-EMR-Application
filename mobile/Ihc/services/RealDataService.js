// This file should hold all the fetch() calls to the Express server
// and the local database calls to the Realm DB
// TODO: Keep realm in sync with mongo
// All lastUpdated fields on mobile-side should be handled within this file.

import Patient from '../models/Patient';
import Status from '../models/Status';
import Soap from '../models/Soap';
import Triage from '../models/Triage';
import DrugUpdate from '../models/DrugUpdate';
import Settings from '../models/Settings';

import {createPatientHelper} from './functions/CreatePatient';
import {signinPatientHelper} from './functions/SigninPatient';
import {updateStatusHelper} from './functions/UpdateStatus';
import {updateSoapHelper} from './functions/UpdateSoap';
import {createDrugUpdateHelper} from './functions/CreateDrugUpdate';
import {getSoapHelper} from './functions/GetSoap';
import {getTriageHelper} from './functions/GetTriage';
import {getPatientHelper} from './functions/GetPatient';
import {getPatientSelectRowsHelper} from './functions/GetPatientSelectRows';
import {updateTriageHelper} from './functions/UpdateTriage';
import {getMedicationUpdatesHelper} from './functions/GetMedicationUpdates';
import {uploadUpdatesHelper} from './functions/UploadUpdates';
import {downloadUpdatesHelper} from './functions/DownloadUpdates';

import Realm from 'realm';
import {stringDate} from '../util/Date';
import config from '../config.json';

// Must set the fetchUrl to the server's IP Address and Port
const fetchUrl = config.fetchUrl;

const realm = new Realm({
  schema: [Patient, Status, Soap, Triage, DrugUpdate, Settings],
  deleteRealmIfMigrationNeeded: true, // TODO: delete when done with dev
});

export function createPatient(patient) {
  return createPatientHelper(patient, realm, fetchUrl);
}

// Check that patient exists, and if so then create a status object for them
export function signinPatient(patientForm) {
  return signinPatientHelper(patientForm, realm, fetchUrl);
}

// field: Any of fields from Status.schema
// value: should match the type that the Status.schema says
export function updateStatus(patientKey, strDate, field, value) {
  return updateStatusHelper(patientKey, strDate, field, value, realm, fetchUrl);
}

export function createDrugUpdate(update) {
  return createDrugUpdateHelper(update, realm, fetchUrl);
}

export function updateSoap(update) {
  return updateSoapHelper(update, realm, fetchUrl);
}

// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
  return getSoapHelper(patientKey, strDate, realm, fetchUrl);
}

export function updateTriage(update) {
  return updateTriageHelper(update, realm, fetchUrl);
}

// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
  return getTriageHelper(patientKey, strDate, realm, fetchUrl);
}

export function getPatient(patientKey) {
  return getPatientHelper(patientKey, realm, fetchUrl);
}

/*
 * Return the statuses of the patients that are active and for this date
 */
export function getPatientSelectRows() {
  return getPatientSelectRowsHelper(realm, fetchUrl);
}

export function getMedicationUpdates(patientKey) {
  return getMedicationUpdatesHelper(patientKey, realm, fetchUrl);
}

export function uploadUpdates() {
  return uploadUpdatesHelper(realm, fetchUrl);
}

export function downloadUpdates() {
  return downloadUpdatesHelper(realm, fetchUrl);
}
