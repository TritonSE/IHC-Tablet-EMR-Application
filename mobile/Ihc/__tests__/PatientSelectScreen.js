import 'react-native';
import React from 'react';
import PatientSelectScreen from '../screens/PatientSelectScreen';

import Patient from '../models/Patient';
import Status from '../models/Status';
import {localData} from '../services/DataService';

import sinon from 'sinon';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  sinon.useFakeTimers(100);
  const patient = Patient.getInstance();
  const patient2 = Patient.getInstance();
  const statuses = [Status.newStatus(patient), Status.newStatus(patient2)];

  sinon.stub(localData, 'getStatuses').returns(statuses);

  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <PatientSelectScreen navigator={fakeNavigator} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
