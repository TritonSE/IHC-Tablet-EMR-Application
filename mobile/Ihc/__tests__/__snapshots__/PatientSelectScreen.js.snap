// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select a Patient
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "backgroundColor": "#F5FCFF",
            "flex": 0,
            "minWidth": "80%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 32,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Name
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Birthday
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Checkin
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Triage
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Doctor
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Pharmacy
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "minWidth": 64,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Notes
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 32,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  firstname father mother
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01/01/00
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  16:0
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              />
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 32,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  firstname father mother
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  01/01/00
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  16:0
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "minWidth": 64,
                  }
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
